### PR TITLE
Fix a few bugs + choice shuffling

### LINKED ISSUE #753: Add ability to suppress choice shuffling in Ordering, CATA andMCQ

### LINKED ISSUE #955: Ordering question should allow authors to disable response randomization/shuffle
**Describe the solution you'd like**
Ordering question should allow authors to disable response randomization/shuffle

### LINKED ISSUE #975: Course projects card view styling is broken
**Describe the bug**
Course projects card view styling is broken

**To Reproduce**
1. Go to authoring workspace, Projects

**Expected behavior**
Cards should be aligned and styled correctly. Cards should use bootstrap card layout for consistency.

**Screenshots**
Screen Shot 2021-04-14 at 12 19 21 PM

**Environment (please complete the following information):**
 - OS: macOS 11.2.3
 - Browser Chrome
 - Version 89

### LINKED ISSUE #992: Internal links do not work in preview
**Describe the bug**
Internal links do not work in preview. The currently just append # to the url.

**Expected behavior**
Internal links should work in preview by linking to the correct url.

### LINKED ISSUE #1012: Empty active input fields on help page look strange
**Describe the bug**

Input fields on the [help](https://proton.oli.cmu.edu/help) page that are active but empty look strange on a current Firefox 88.0.

**Screenshots**

empty input field:

empty input field

non-empty input field:

non-empty input field

**Environment (please complete the following information):**
 - OS: Fedora 32
 - Browser: Firefox
 - Version: 88.0

### PR DESCRIPTION
**Pull request by ThinkThoughtThunk**
I've refactored a large part of the client-side activity implementations but added some bug fixes into that branch that I'm still working through so I've pulled out the choice shuffling and a couple of the finished fixes for a PR.

Enhancements:
- Add support for disabling answer choice shuffling in multiple choice, check all that apply, ordering questions

Bug fixes:
- Fix internal links in page preview mode
- Fix projects view project card styling
- Fix problem with inputs causing clipping in Firefox

Closes #753 
Closes #955 
Closes #975 
Closes #992 
Closes #1012

### CODE REVIEW

File: assets/src/components/activities/common/utils.tsx [hunk 1]
```diff
@@ -17,3 +18,47 @@ export function fromText(text: string): { id: string, content: RichText } {
     },
   };
 }
+
```
Comment (LEDThereBeLight) [inline assets/src/components/activities/common/utils.tsx:21]: Answer choice shuffling

File: lib/oli/authoring/editing/page_editor.ex [hunk 2]
```diff
@@ -240,7 +240,8 @@ defmodule Oli.Authoring.Editing.PageEditor do
          render_context <- %Rendering.Context{
```
Comment (LEDThereBeLight) [inline lib/oli/authoring/editing/page_editor.ex:240]: Internal link fix

File: lib/oli_web/live/projects/card.ex [hunk 3]
```diff
@@ -4,13 +4,17 @@ defmodule OliWeb.Projects.Card do
...
   def render(assigns) do
     ~L"""
-    <a href="<%= Routes.project_path(OliWeb.Endpoint, :overview, @project)%>" class="card-link">
-      <div class="card-link">
-        <h1><%= @project.title  %></h1>
-        <p><%= @project.description %></p>
-        <span class="authors"><%= @author_count %> <i class="material-icons-outlined">person</i>
-      </div>
-    </a>
+    <div class="col my-1">
```
Comment (LEDThereBeLight) [inline lib/oli_web/live/projects/card.ex:7]: Projects view card styling fix

**Review by darrensiegel — APPROVED** [review]
Nice work, and great catch to update those placeholder texts

**Pull request merged**